maas-api/internal/token/manager.go

```diff
@@ -97,13 +125,23 @@ func (m *Manager) RevokeTokens(ctx context.Context, user *UserContext) error {
 	_, err = m.serviceAccountLister.ServiceAccounts(namespace).Get(saName)
 	if errors.IsNotFound(err) {
 		log.Printf("Service account %s not found in namespace %s, nothing to revoke", saName, namespace)
+		// Still try to delete token metadata even if SA doesn't exist
+		if err := m.store.DeleteTokensForUser(ctx, namespace, user.Username); err != nil {
+			log.Printf("Failed to delete tokens for user %s: %v", user.Username, err)
+		}
```

Comment:
If we have new `api_keys` service/manager we could do this call there instead of mixing two responsibilities here.